lock/src/main/java/com/auth0/android/lock/Lock.java

```diff
@@ -376,5 +372,31 @@ public Builder withProviderResolver(@NonNull AuthProviderResolver resolver) {
             ProviderResolverManager.set(resolver);
             return this;
         }
+
+        /**
+         * Displays a second screen with the specified custom fields during sign up.
+         * Each field must have a unique key.
+         *
+         * @param customFields the custom fields to display in the sign up flow.
+         * @return the current builder instance
+         */
+        public Builder withSignUpFields(List<CustomField> customFields) {
+            removeDuplicatedKeys(customFields);
+            options.setCustomFields(customFields);
+            return this;
+        }
+
+        private void removeDuplicatedKeys(List<CustomField> customFields) {
+            //Remove duplicated keys
+            Map<String, CustomField> map = new LinkedHashMap<>();
```

Comment:
I'd rather iterate the list of fields and check if I already have one with the same key (e.g. using a `Set`).